salt/cloud/clouds/vmware.py

```diff
@@ -230,7 +230,7 @@ def _edit_existing_hard_disk_helper(disk, size_kb):
     return disk_spec
 
 
-def _add_new_hard_disk_helper(disk_label, size_gb, unit_number, controller_key=1000):
+def _add_new_hard_disk_helper(disk_label, size_gb, unit_number, thin_provision=False, controller_key=1000):
```

Comment:
Can you move the `thin_provision` option to after the `controller_key` kwarg here? We like to try and avoid API changes where possible, just in case people are using positional kwargs somewhere, we don't what things to break.